### ISSUE TITLE
`dir_fd=None` blocks PTH106, PTH107, PTH108, and PTH115

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#21342]
### Summary

[`os-rmdir` (PTH106)](https://docs.astral.sh/ruff/rules/os-rmdir/), [`os-remove` (PTH107)](https://docs.astral.sh/ruff/rules/os-remove/), [`os-unlink` (PTH108)](https://docs.astral.sh/ruff/rules/os-unlink/), and [`os-readlink` (PTH115)](https://docs.astral.sh/ruff/rules/os-readlink/) have false negatives when `dir_fd` is explicitly set to its default value of `None`. All other PTH rules for functions that support `dir_fd` handle `dir_fd=None`.
```console
$ cat >pth1.py <<'# EOF'
import os
os.rmdir("path", dir_fd=None)
os.remove("path", dir_fd=None)
os.unlink("path", dir_fd=None)
os.readlink("path", dir_fd=None)
# EOF

$ ruff --isolated check pth1.py --select PTH106,PTH107,PTH108,PTH115
All checks passed!
```

### Version

ruff 0.14.4 (c7ff9826d 2025-11-06)
